packages/ui/components/head-seo/HeadSeo.tsx

```diff
@@ -72,13 +73,15 @@ const buildSeoMeta = (pageProps: {
 };
 
 export const HeadSeo = (props: HeadSeoProps): JSX.Element => {
-  const defaultUrl = getBrowserInfo()?.url;
+  // build the canonical url to ensure it's always cal.com (not app.cal.com)
+  const router = useRouter();
+  const calcomUrl = (`https://cal.com` + (router.asPath === "/" ? "" : router.asPath)).split("?")[0]; // cut off search params
+  const defaultUrl = IS_SELF_HOSTED ? getBrowserInfo()?.url : calcomUrl;
```

Comment:
we can use isCalcom here https://github.com/calcom/cal.com/blob/main/packages/lib/isCalcom.ts

Reply:
Updated to use it [here](https://github.com/calcom/cal.com/pull/6693/commits/2f47eb2c09c32227df38ad523f929881ed9e865f).

I think we could still need the `(router.asPath === "/" ? "" : router.asPath)` though as we still have to omit the trailing slash on the homepage?